IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormPushTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CommonTestUtils;
+using FluentAssertions;
+using GitCommands.Git;
+using GitUI;
+using GitUI.CommandsDialogs;
+using NUnit.Framework;
+
+namespace GitExtensions.UITests.CommandsDialogs
+{
+    [Apartment(ApartmentState.STA)]
+    public class FormPushTests
+    {
+        // Created once for the fixture
+        private ReferenceRepository _referenceRepository;
+
+        // Created once for each test
+        private GitUICommands _commands;
+
+        [SetUp]
+        public void SetUp()
+        {
+            if (_referenceRepository == null)
+            {
+                _referenceRepository = new ReferenceRepository();
+            }
+            else
+            {
+                _referenceRepository.Reset();
+            }
+
+            _commands = new GitUICommands(_referenceRepository.Module);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            _referenceRepository.Dispose();
+        }
+
+        [TestCase(false, false, false, ForcePushOptions.DoNotForce)]
+        [TestCase(false, true, true, ForcePushOptions.Force)]
+        [TestCase(false, true, false, ForcePushOptions.DoNotForce)] // ForcePushBranches state is bound to ForcePushTags state
+        [TestCase(false, false, true, ForcePushOptions.Force)] // ForcePushTag requires normal force as with-lease is not allowed for tags
+        [TestCase(true, false, false, ForcePushOptions.ForceWithLease)]
+        [TestCase(true, true, false, ForcePushOptions.ForceWithLease)] // ForcePushBranches state is bound to ForcePushTags state
+        [TestCase(true, false, true, ForcePushOptions.Force)] // ForcePushBranches and ForcePushTags take precedence over ckForceWithLease
+        [TestCase(true, true, true, ForcePushOptions.Force)] // ForcePushBranches and ForcePushTags take precedence over ckForceWithLease
```

Comment:
@RussKie 
When I changed `public CheckBox ForcePushBranches() => _form.ForcePushTags;` to `public CheckBox ForcePushBranches => _form.ForcePushBranches;`, I got some test failures, as I had expected.  When fixing up the test cases and reviewing them once they all were passing, I had discovered that it seems like databinding is not actually working in this test.  I expected that if you had `forcePushBranchWithLeaseChecked = true`, `forcePushBranchChecked = false`, `forcePushTagChecked = false`, we would get the result of `ForcePushOptions.DoNotForce`, since setting the force push tag to false should have resulted in force push branch with lease to be false...

I did a little bit of searching on the internet to see how to resolve this and came across this [StackOverflow answer](https://stackoverflow.com/a/10040854) and tried a few solutions from the linked answer's comments but could not get databinding to work in the test.  I will remove the comments about bound state and make a comment, something like that the databinding is not working for the test and that the expected test case results reflect this.